src/test/kotlin/com/ampnet/identityservice/service/BlockchainQueueServiceTest.kt

```diff
@@ -89,43 +97,134 @@ class BlockchainQueueServiceTest : TestBase() {
         suppose("Blockchain service will whitelist any address") {
             given(blockchainService.whitelistAddress(any(), any(), any())).willReturn(hash)
         }
+
         suppose("Transactions are mined") {
             given(blockchainService.isMined(any(), any())).willReturn(true)
         }
+
         suppose("There are two tasks in queue") {
             testContext.task = createBlockchainTask()
             createBlockchainTask()
         }
 
         verify("Service will handle the task") {
-            waitUntilTasksAreProcessed()
+            processAllTasks()
+
             val tasks = blockchainTaskRepository.findAll()
+
             assertThat(tasks).hasSize(2)
             assertThat(tasks).allMatch { it.status == BlockchainTaskStatus.COMPLETED && it.hash == hash }
         }
     }
 
+    @Test
+    fun mustFailPendingTaskWhenExceptionIsThrown() {
+        suppose("Blockchain service will throw exception") {
+            given(blockchainService.whitelistAddress(any(), any(), any())).willThrow(RuntimeException())
+        }
+
+        suppose("There is a task in queue") {
+            createBlockchainTask(status = BlockchainTaskStatus.CREATED)
+        }
+
+        verify("Task will fail") {
+            processAllTasks(maxAttempts = 1)
+
+            val tasks = blockchainTaskRepository.findAll()
+
+            assertThat(tasks).hasSize(1)
+            assertThat(tasks.first().status).isEqualTo(BlockchainTaskStatus.FAILED)
+        }
+    }
+
+    @Test
+    fun mustFailInProcessTaskWhenExceptionIsThrown() {
+        suppose("Blockchain service will throw exception") {
+            given(blockchainService.isMined(hash, chainId)).willThrow(RuntimeException())
+        }
+
+        suppose("There is a task in process") {
+            createBlockchainTask(status = BlockchainTaskStatus.IN_PROCESS, hash = hash)
+        }
+
+        verify("Task will fail and new task will be created") {
+            processAllTasks(maxAttempts = 1)
+
+            val tasks = blockchainTaskRepository.findAll()
+
+            assertThat(tasks).hasSize(1)
+            assertThat(tasks.first().status).isEqualTo(BlockchainTaskStatus.FAILED)
+            assertThat(tasks.first().hash).isEqualTo(hash)
+        }
```

Comment:
Looks like verify description is incorrect because there is only one task in repo.